frontend/src/main/angular/src/app/gog/components/modals/gog-auth-modal/gog-auth-modal.component.ts

```
@@ -0,0 +1,108 @@
+
+  private extractCodeFromUrl() {
+    const gogCodeUrl = this.gogCodeUrlInput.value;
+    const params: URLSearchParams = new URL(gogCodeUrl).searchParams;
+    return params.get("code") as string;
+  }
+
+  getFormErrors(form: FormGroup): Record<string, any> {
+    return Object.entries(form.controls)
+      .filter(([_, control]) => control.errors)
+      .reduce((errorMap, [controlName, control]) => ({
+        ...errorMap,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)